### PR TITLE
Patch 25 - Furthest pair

### PR DESCRIPTION
**Pull request by wtaylor17**
Solves https://open.kattis.com/problems/roberthood but has good enough time complexity for more intense problems as well

### CODE REVIEW

File: snippets/2D_geometry/furthest_pair.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// returns diameter (max pairwise distance) of points on a convex hull.
+// needs that all points are distinct and none are collinear
+// takes time linear in the size of the hull
+pair<point,point> hull_diameter(convex_polygon poly) {
+    vector<point> &S = poly.points;
+    if(S.size() < 2) return {point(), point()};
+    int n = S.size(), j = 1;
+    pair<ll, pair<point,point>> res({0, {S[0], S[0]}});
+    for(int i=0;i<j;i++)
+        for (;; j = (j + 1) % n) {
```
Comment (JarateKing) [inline snippets/2D_geometry/furthest_pair.cpp:13]: can this be a binary search instead?

Comment (JarateKing) [reply snippets/2D_geometry/furthest_pair.cpp:13]: hmm, maybe not: consider a case such as

image

where you can't really binary search a function like that

Comment (wtaylor17) [reply snippets/2D_geometry/furthest_pair.cpp:13]: it uses the [method of rotating calipers](https://en.wikipedia.org/wiki/Rotating_calipers)

Comment (JarateKing) [reply snippets/2D_geometry/furthest_pair.cpp:13]: ah, my bad I didn't fully understand the code. That looks good to me

**Pull request merged**